src/FffGcodeWriter.cpp

```diff
@@ -1517,15 +1515,28 @@ bool FffGcodeWriter::processSkinPart(const SliceDataStorage& storage, LayerPlan&
         }
     }
 
-    int extra_infill_shift = 0;
-    coord_t offset_from_inner_skin_infill = 0;
+    Polygons skin_polygons;
+    Polygons skin_lines;
     Polygons concentric_perimeter_gaps; // the perimeter gaps of the insets of concentric skin pattern of this skin part
-    Polygons* perimeter_gaps_output = (fill_concentric_perimeter_gaps)? &concentric_perimeter_gaps : nullptr;
-    Infill infill_comp(pattern, skin_part.inner_infill, offset_from_inner_skin_infill, skin_line_width, skin_line_width, skin_overlap, skin_angle, z, extra_infill_shift, perimeter_gaps_output);
-    infill_comp.generate(skin_polygons, skin_lines);
+
+    const bool process_skin_infill = extruder_nr == top_bottom_extruder_nr;
+    const bool generate_perimeter_gaps = fill_concentric_perimeter_gaps && pattern == EFillMethod::CONCENTRIC; // whether we need to generate perimeter gaps for concentric infill of the skinfill area
+
+    // generate skin_polygons and skin_lines (and concentric_perimeter_gaps if needed)
+    if (process_skin_infill || generate_perimeter_gaps)
+    {
+        // only generate infill if we're going to print it with this extruder
+        // or when we need to compute the perimeter gaps
+        // TODO: only compute this once when the infill is concentric and the perimeter gpas are printed with a different extruder
+        int extra_infill_shift = 0;
+        coord_t offset_from_inner_skin_infill = 0;
```

Comment:
Actually, I have no idea. I can't find this variable any more.
Please comment on final PR, this comment is outdated already.